File: Typescript/test/getSneakyNumbers.test.ts
import { expect, test } from '@jest/globals';
import * as solution from '../src/getSneakyNumbers';

test('Example 1', () => {
	let input = [0, 1, 1, 0];
	let expected = [0, 1];
	let output = solution.getSneakyNumbers(input).sort((a, b) => a - b);
	expect(output).toEqual(expected);
});

test('Example 2', () => {
	let input = [0, 3, 2, 1, 3, 2]
	let expected = [2, 3];
	let output = solution.getSneakyNumbers(input).sort((a, b) => a - b);
	expect(output).toEqual(expected);
});

test('Example 3', () => {
	let input = [7, 1, 5, 4, 3, 4, 6, 0, 9, 5, 8, 2];
	let expected = [4, 5];
	let output = solution.getSneakyNumbers(input).sort((a, b) => a - b);
	expect(output).toEqual(expected);
});

File: Dart/test/getSneakyNumbers_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:leetcode/getSneakyNumbers.dart';

void main() {
  test("test example 1", () {
    var input = [0, 1, 1, 0];
    var expected = [0, 1];

    final sln = Solution();
    var output = sln.getSneakyNumbers(input);
    output.sort();
    expected.sort();
    expect(output, expected);
  });
  test("test example 2", () {
    var input = [0, 3, 2, 1, 3, 2];
    var expected = [2, 3];

    final sln = Solution();
    var output = sln.getSneakyNumbers(input);
    output.sort();
    expected.sort();
    expect(output, expected);
  });
  test("test example 3", () {
    var input = [7, 1, 5, 4, 3, 4, 6, 0, 9, 5, 8, 2];
    var expected = [4, 5];

    final sln = Solution();
    var output = sln.getSneakyNumbers(input);
    output.sort();
    expected.sort();
    expect(output, expected);
  });
}
